components/ReleaseStats/ReleaseStats.jsx

```diff
@@ -0,0 +1,142 @@
+import { useState } from "react"
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs"
+import {
+  Dialog,
+  DialogTrigger,
+  DialogContent,
+  DialogPortal,
+  DialogClose,
+} from "@/components/Dialog/Dialog"
+
+let today = new Date()
+
+let yesterday = new Date(today)
+yesterday.setDate(yesterday.getDate() - 1)
+
+let tomorrow = new Date(today)
+tomorrow.setDate(tomorrow.getDate() + 1)
+
+let pastWeek = new Date(today)
+pastWeek.setDate(pastWeek.getDate() - 6)
+
+let pastTwoWeeks = new Date(today)
+pastTwoWeeks.setDate(pastTwoWeeks.getDate() - 13)
+
+let past30Days = new Date(today)
+past30Days.setDate(past30Days.getDate() - 29)
```

Comment:
What if you had a function that you could pass the amount of days ago into?

```js
let now = new Date()

function getPastDate(days) {
  let date = new Date(now)
  date.setDate(date.getDate() - days)
  
  return date.toLocaleDateString();
}

const statsFrom = [
  {
    label: "Today",
    start: now.toLocaleDateString(),
  },
  {
    label: "Yesterday",
    start: getPastDate(1),
    end: getPastDate(1)
  },
  {
    label: "Last 7 Days",
    start: getPastDate(7),
  }
  // and so on
]
```

Then in your jsx:

```jsx
{statsFrom.map((date) => (
  <button
    key={data.label}
    className="button"
    data-variant="primary"
    onClick={() => getCodes(date.start, date.end ?? now.toLocaleDateString())}
  >
    {date.label}
  </button>
))}
```

Reply:
Here's context on why you shouldn't set `index` to the `key` in a loop: https://legacy.reactjs.org/docs/lists-and-keys.html#:~:text=We%20don't%20recommend%20using,an%20index%20as%20a%20key.